# Computing Fibonacci n-th term.
## By: Jhonatan Rivera Saumeth.

We are going to use `sys` module to return the max value for an `int` type variable.

In [ ]:
import sys
#Create variables and assign their initial values.
IntFn_2=0
IntFn_1=1
IntFn=0

Try to compute 100-th term of fibonacci sequence, if n-th term is greater than max `int` value it prints (n-1)-th.

In [ ]:
n= 100
overFlow=0
for i in range(n):
    IntAux=IntFn 
    IntFn_2=IntFn_1
    IntFn_1=IntFn
    IntFn=IntFn_2+IntFn_1
    #--------------------
    if(IntFn>sys.maxint and overFlow==0):
        print ("Overflow computing "+str(i)+"-th term of Fibonacci Sequence(data type: Int) ");
        overFlow=1


Python has two types of integer values, `int` and `long`, since [long does not have bounds](https://docs.python.org/2.4/lib/typesnumeric.html), we cant find an overflow using `long` data type.

let's discover what is the max term.

In [9]:
import sys
#Create variables and assign their initial values.
IntFn_2=0
IntFn_1=1
IntFn=0
n= 100
overFlow=0
for i in range(n):
    IntAux=IntFn 
    IntFn_2=IntFn_1
    IntFn_1=IntFn
    IntFn=IntFn_2+IntFn_1
    #--------------------
    if(IntFn>sys.maxint and overFlow==0):
        print ("Overflow computing "+str(i+1)+"-th term of Fibonacci Sequence(data type: Int) ");
        print(str(i)+"-th term is: "+str(IntFn_1)+" and the next one is: "+str(IntFn))
        overFlow=1
        print ("Max int: "+str(sys.maxint))

Overflow computing 47-th term of Fibonacci Sequence(data type: Int) 
46-th term is: 1836311903 and the next one is: 2971215073
Max int: 2147483647


As you can see max int is: "2147483647" (32-bit Signed int), after it
python start to work with long values, we can verify it.

In [12]:
import sys
IntorLong = sys.maxint
print type(IntorLong)
IntorLong=IntorLong+1
print type(IntorLong)

<type 'int'>
<type 'long'>
